pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/TeiXml.scala

```diff
@@ -0,0 +1,47 @@
+package weco.pipeline.transformer.tei
+
+import scala.util.Try
+import scala.xml.{Elem, XML}
+
+class TeiXml(xml: Elem) {
+  val id: String = xml.attributes
+    .collectFirst {
+      case metadata if metadata.key == "id" => metadata.value.text
+    }
+    .getOrElse(throw new RuntimeException(s"Could not find an id in XML!"))
+
+  def bNumber: Either[Throwable, Option[String]] = {
+    val identifiersNodes = xml \\ "msDesc" \ "msContents" \ "msIdentifier" \ "altIdentifier"
+    val seq = (identifiersNodes.filter(
+      n => (n \@ "type").toLowerCase == "sierra"
+    ) \ "idno").toList
+    seq match {
+      case List(node) => Right(Some(node.text.trim))
+      case Nil        => Right(None)
+      case _          => Left(new RuntimeException("More than one sierra bnumber node!"))
+    }
+  }
```

Comment:
Could you provide an example of the XML you're trying to parse here? Kinda tricky to know if this code is correct without seeing it. Look at the METS transformer for examples.

Is it worth logging the nodes we got if there was more than one?